src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/FilterDelegationBase.cs

```diff
@@ -64,7 +64,10 @@ public override bool TryGetDelegationMetadata(CallNode node, TexlBinding binding
             return true;
         }
 
-        protected bool IsValidDelegatableFilterPredicateNode(TexlNode dsNode, TexlBinding binding, FilterOpMetadata filterMetadata, bool generateHints = true)
+        // Determine whether a node can be delegated as part of a filter predicate.
+        // The enforeceBoolean flag determines whether to enforce the return type of the node.  If the node is part of a filter predicate directly, it must return a boolean type.
+        // If the node is used in other places, such as a LookUp reduction formula inside a filter, it can return any type.
```

Comment:
>LookUp reduction formula inside a filter [](http://example.com/codeflow?start=58&length=40)

Suggestion: '... such as a reduction / projection formula inside a LookUp'. There is no reduction formula in a filter, only in a lookup.

Reply:
Clarified the scenario in the comment